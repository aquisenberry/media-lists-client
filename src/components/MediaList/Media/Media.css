.media {
  width: 100%;
  height: 100%;
  margin: 0;
  border-radius: 12px;
  /* overflow: hidden; */
  /* border: none; */
  /* transition: all 0.4s cubic-bezier(0.175, 0.885, 0, 1); */
  box-shadow: 0px 13px 10px -7px rgba(0, 0, 0, 0.1);
}
.media:hover {
  box-shadow: 0px 30px 18px -8px rgba(0, 0, 0, 0.1);
  transform: scale(1.05, 1.05);
}

.media__year-container {
  position: absolute;
  padding: 16px;
  width: 100%;
  opacity: 0;
  top: 0;
  color: #f9d3b4;
}

.media__image-container {
  width: 100%;
  height: 100%;
}

.media__image {
  margin: 1em 5px 17px 0;
  max-height: 250px;
}
.media__meta-container {
  z-index: 2;
  background-color: #343739;
  padding: 16px 24px 24px 24px;
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
}
.media__meta-type {
  font-family: "Raleway", sans-serif;
  text-transform: uppercase;
  font-size: 13px;
  letter-spacing: 2px;
  font-weight: 500;
  color: #f0f0f0;
}

.media__meta-title {
  margin-top: 5px;
  font-family: "Roboto Slab", serif;
  color: #f9d3b4;
}

.media:hover .media__image-container {
  height: 100%;
  opacity: 0.3;
}
.media:hover .media__meta-container {
  background-color: transparent;
}
.media:hover .media__year-container {
  opacity: 1;
}
