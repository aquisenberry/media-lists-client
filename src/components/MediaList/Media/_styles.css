.media {
  /* width: 100%; */
  height: 100%;
  margin: 0;
  border-radius: 12px;
  /* overflow: hidden; */
  /* border: none; */
  /* transition: all 0.4s cubic-bezier(0.175, 0.885, 0, 1); */
  box-shadow: 0px 13px 10px -7px rgba(0, 0, 0, 0.1);
  position: relative;
}
.media:hover {
  box-shadow: 0px 30px 18px -8px rgba(0, 0, 0, 0.2);
  transform: scale(1.05, 1.05);
}
.media-meta {
  position: absolute;
  top: 0;
  z-index: 1;
  height: 100%;
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}
.media-meta__year {
  /* position: absolute; */
  padding: 16px;
  width: 100%;
  opacity: 0;
  top: 0;
  color: #f9d3b4;
  padding: 1.5rem 1rem;
  text-align: right;
}
.media-meta__title {
  margin-top: 5px;
  opacity: 0;
  padding: 0 1rem;
}

.media__image-container {
  width: 100%;
  height: 100%;
}

.media__image {
  margin: 0 5px 17px 0;
  max-height: 250px;
}
.media__meta-container {
  z-index: 2;
  background-color: #343739;
  padding: 16px 24px 24px 24px;
  position: absolute;
  bottom: 0;
  right: 0;
  left: 0;
}
.media-meta__type {
  font-family: "Raleway", sans-serif;
  text-transform: uppercase;
  font-size: 13px;
  letter-spacing: 2px;
  font-weight: 500;
  color: #f0f0f0;
}

.media:hover .media__image-container {
  height: 100%;
  opacity: 0.3;
}
.media:hover .media-meta {
  background-color: transparent;
}
.media:hover .media-meta__year,
.media:hover .media-meta__title {
  opacity: 1;
}
