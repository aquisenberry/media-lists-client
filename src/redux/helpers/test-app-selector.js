
const state = {
    external: {
        searchTerm: "",
        movies:[],
        shows:[],
        videoGames: [],
        boardGames: [],
        books: [],
        mediaDetails: {}
    },
    page: {
        media: [],
        contextMenuAnchor: {
            x:0,
            y:0
        },
        showContextMenu: false,
        contextTarget:{},
        contextType: false
    }
}

export const testUseAppSelector = (f) => f(state)