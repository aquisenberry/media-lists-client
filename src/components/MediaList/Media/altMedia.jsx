import React from 'react'
import './Media.css'

const Media = ({media}) => {
    return (
        <div className="media">
            <div className="media__image-container">
                <img 
                    className="media__image"
                    src={media.poster ? media.poster : 'https://via.placeholder.com/400'}
                    alt={media.poster ? media.title + ' Poster' : 'placeholder image'}
                />
            </div>
        </div>
    )

}

export default Media