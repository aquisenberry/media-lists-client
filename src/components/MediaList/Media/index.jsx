/*
    TODO: implement poster minimum height
*/

import React, {useEffect} from 'react'
import  { addMedia, enterMedia, exitMedia } from '../../../redux/actions/page'
import { useAppDispatch } from '../../../redux/helpers/redux-hooks'
import './_styles.css'

const Media = ({media}) => {
    const dispatch = useAppDispatch()
    useEffect(() => {
        dispatch(addMedia({...media, selected: false}))
    }, [dispatch])

    const handleMouseEnterMedia = (e) => {
        dispatch(enterMedia(media))
    }
    const handleMouseExitMedia = (e) => {
        dispatch(exitMedia())
    }
    return (
        <div className="media" onMouseEnter={handleMouseEnterMedia} onMouseLeave={handleMouseExitMedia}>
            <div className="media__image-container">
                <img 
                    className="media__image"
                    src={media.poster}
                    alt={`${media.title} Poster`}
                />
            </div>
            <div className="media-meta">
                <h3 className="media-meta__title">{media.title}</h3>
                <h4 className="media-meta__year">{media.year}</h4>
            </div>
        </div>
    )

}

export default Media