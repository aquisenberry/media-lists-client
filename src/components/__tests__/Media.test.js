import renderer from 'react-test-renderer'
import Media from '../MediaList/Media'
import {useAppDispatch, useAppSelector} from '../../redux/helpers/redux-hooks'
import { testUseAppSelector } from '../../redux/helpers/test-app-selector'
import {media} from './__mocks__/Media.mocks.json'

jest.mock('../../redux/helpers/redux-hooks', () =>({
    useAppDispatch: jest.fn(() => {}),
    useAppSelector: jest.fn()
}))

describe("Media", () =>{
    
    beforeEach(() =>{
        const dispatch = jest.fn();
        useAppDispatch.mockReturnValue(dispatch);
        useAppSelector.mockImplementation(testUseAppSelector)
    })
    afterEach(() =>{
        jest.clearAllMocks()
    })
    it('has no markup regressions', () =>{
        const component = renderer.create(
            <Media media={media} />
        )

        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
        
    })
    it('renders expected values', () => {
        const component = renderer.create(
            <Media media={media} />
        )
        const instance = component.root


        expect(instance.findByProps({className: 'media__image'}).props.src).toEqual('http://image.tmdb.org/t/p/w185/wSqAXL1EHVJ3MOnJzMhUngc8gFs.jpg')
        expect(instance.findByProps({className: 'media__image'}).props.alt).toEqual('Orphan: First Kill Poster')
        expect(instance.findByProps({className: 'media-meta__title'}).children).toContain('Orphan: First Kill')
        expect(instance.findByProps({className: 'media-meta__year'}).children).toContain('2022')
    })

})
