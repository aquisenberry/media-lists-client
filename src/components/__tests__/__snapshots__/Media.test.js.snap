// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Media has no markup regressions 1`] = `
<div
  className="media"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="media__image-container"
  >
    <img
      alt="Orphan: First Kill Poster"
      className="media__image"
      src="http://image.tmdb.org/t/p/w185/wSqAXL1EHVJ3MOnJzMhUngc8gFs.jpg"
    />
  </div>
  <div
    className="media-meta"
  >
    <h3
      className="media-meta__title"
    >
      Orphan: First Kill
    </h3>
    <h4
      className="media-meta__year"
    >
      2022
    </h4>
  </div>
</div>
`;
